### DISCUSSION TITLE
Mining pool.

### DISCUSSION THREAD

**Discussion by Piramida256** [CyberChainXyz/go-cyberchain#2] [General]
When pool?

**Comment by 3dpasspool** [comment]
Is there a tutorial for creating pools?

**Reply by nathanccxv** [reply]
If you have experience in software engineering and mining algorithms, https://github.com/etclabscore/open-etc-pool is a great starting point.

1. CyberChain uses fphash PoW, so you'll need to modify the block validation algorithm. You can find the fphash algorithm here: https://github.com/CyberChainXyz/fphash-go.

2. If you want users to mine with xMiner (https://github.com/CyberChainXyz/xMiner), you should look into the stratum-jsonrpc2.0-ws protocol (https://github.com/CyberChainXyz/stratum-jsonrpc2-ws), which is the protocol for interaction between xMiner and the pool.

**Comment by Alex-007-Alex** [comment]
Can't connect with hive

**Reply by Piramida256** [reply]
Same for me

**Reply by nathanccxv** [reply]
I'm not familiar with hive, could you describe the problem you're encountering in detail?

**Comment by Piramida256** [comment]
How connect with hive? Need pool!

**Comment by StianNOR** [comment]
Hive work perfectly but you cant use the GUI. Need to ssh and download miner and node. But yeah we need pool high diff

**Reply by Alex-007-Alex** [reply]
I kant connect with hive to 127.0.0.....
With windows without problems

**Reply by Alex-007-Alex** [reply]
![Screenshot_20241222_091915_Gallery](https://github.com/user-attachments/assets/4c95b43f-7a2f-4788-969a-a65bf322a548)

**Reply by Alex-007-Alex** [reply]
Node begin run in Hive, bu stop on this moment

**Reply by Alex-007-Alex** [reply]
I'm using ssh
I know how do it;) you have instructions  but problems that node running 10 second and stop on moment

**Reply by Alex-007-Alex** [reply]
https://github.com/user-attachments/assets/50d42260-8fd3-43a7-93c4-4a9cb56e2767

Not perfectly 😕

**Reply by nathanccxv** [reply]
This issue could be a bug in go-cyberchain that prevents it from running on a CPU that doesn't support AVX2. Unfortunately, the only current solution is to run the CyberChain node on another computer that supports AVX2.

You can try running the node on another computer.
Additionally, you will need to modify the **run_for_mining.sh** file by changing:

`./ccx -ws -mine -miner.etherbase=0x...`
to:
`./ccx -ws -ws.addr=0.0.0.0 -mine -miner.etherbase=0x...`

For xMiner, in the **solo_mining.sh** file, replace **127.0.0.1** with the IP address of the computer where the node is running:

`./xMiner -pool=ws://**your-node-ip**:8546`

Could you also paste the result of the **lscpu** command here so we can confirm if your CPU supports AVX2?

**Reply by Alex-007-Alex** [reply]
No,it's not working. I know how it must be.

I have node on my pc with cpu 13900k, and mining with 3080 running there, but can't connect to the node in my local net. With another coins without problems I can do it, but can't with ccx.
The ports in my router 30303 automatically weired, port 8564 is triggered and open, but not working. 
Some people have same problems.

**Reply by Alex-007-Alex** [reply]
./ccx -ws -mine -miner.etherbase=0x...
к:
./ccx -ws -ws.addr=0.0.0.0 -mine -miner.etherbase=0x

I will Try to change it

**Reply by Alex-007-Alex** [reply]
Error starting protocol stack: listen tcp 192.168.0.107:8546: bind: cannot assign requested address

**Reply by nathanccxv** [reply]
Please check if multiple instances of ccx are running on the same machine. **It’s best to avoid this.**

If you do need to run multiple instances, you’ll need to change the default ws port. You can add the -ws.port=8547. Additionally, make sure to modify the default -port and -datadir parameters as well.

**Comment by coolpooltop** [comment]
![image](https://github.com/user-attachments/assets/6837a65d-a866-41c0-a30a-1d404f987b9e)
**https://coolpool.top**

**Reply by BkewlBro** [reply]
To run the pool:
CCX Must be closed!(normally used when solo mining)

Download and Unzip [Xminer](https://github.com/CyberChainXyz/xMiner/releases/tag/v0.1.15-beta)
Edit File "pool_mining.bat" and change:

username = PayoutAddress.WorkerName (worker name optional)
wss://pool-address.com:port = "PoolAddress:port"

Example:
xMiner.exe -user=0xC1f8945259241f92c1817b3dc175358ce313c973.Miner123 -pass=password -pool=ws://ccx.coolpool.top:14003
pause

**Reply by qusai1991234** [reply]
can we use hivos ??

**Reply by BkewlBro** [reply]
> can we use hivos ??

To be honest I've never used Hivos, I don't know if it supports the "fphash" algorithm or not, but if you can download and intall the repo for the algo, I'm sure it would. Otherwise Xminer seems to be other only miner I have looked though that supports the algorithm.

**Reply by StianNOR** [reply]
> can we use hivos ??

Yea but you cant use the GUI. you can only use GUI for setting overclock and look at temp and power usage. you need to start it in ssh. Google is your friend.
